sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -333,6 +371,9 @@ private PlannerResult planWithDruidConvention(
                           .filter(action -> action.getAction() == Action.READ)
                           .collect(Collectors.toSet());
 
+        // TODO: This is not really a state check since there is a race condition.
```

Comment:
What's the TODO here — I don't see what change needs to be made later?

And, whatever that change is, why not make it now?

Reply:
That rework suggested that there are several other areas to clean up, and that `SqlLifecycleTest` should be rewritten. That will come later, to limit the blast radius of this PR.